### DISCUSSION TITLE
Lesson 3: Fund Me

### DISCUSSION THREAD

**Discussion by vicshi06** [smartcontractkit/full-blockchain-solidity-course-py#770] [Q&A] [answered]
Hi.

When I was going through Lesson 3, I saw that the ` addressToAmountFunded` is followed by a `+=`. However, what is the difference between ` addressToAmountFunded[msg.sender] += msg.value` and ` addressToAmountFunded[msg.sender] = msg.value`? Can someone please explain? Thanks! 

![image](https://user-images.githubusercontent.com/90171690/149256959-d1bd3a04-e51e-4857-9b71-747ac06d0113.png)

**Comment by cromewar** [comment, marked as answer]
Hello @vicshi06 Yeah this is a difference, let's see.

The `=` is for assignation, you are saying that whatever is on the right side should be stored on the left side, example:

`variable a = 50` this meas variable a <- 50 

Quite easy right?

And `+=` means add to the current value, let's suppose we have another variable called `b` which has the value of 5, so if we say:

`b += 1` this means we are saying, add 1 to whatever is on variable b: 5  + 1.

I hope this info might help.

**Reply by vicshi06** [reply]
Okay I understand the examples you gave, but what does the `+=` mean in the context of `addressToAmountFunded[msg.sender] += msg.value`. Does that mean every time someone sends me money, that person is being added to the array? 

If we replace `+=` with `=`, does that mean that person is not being added to the array?

**Reply by cromewar** [reply]
Setting `+=` you are adding both, you are saying that what ever is already there should be added using the new value, if you use `=` you are just saying you are going to replace whatever is in there with the new value.

**Reply by vicshi06** [reply]
Thanks so much!

**Reply by vicshi06** [reply]
One more question actually, does mapping function in solidity automatically create an array?

**Reply by cromewar** [reply]
Hello @vicshi06, sorry for the delay, mappings are it's own type and should not be considered as arrays, check this out for more info: https://medium.com/upstate-interactive/mappings-in-solidity-explained-in-under-two-minutes-ecba88aff96e
